block/meta-iosched.c

```diff
@@ -0,0 +1,75 @@
+#include <linux/blkdev.h>
+#include <linux/elevator.h>
+#include <linux/slab.h>
+
+#include "meta-iosched.h"
+
+struct queue_arr {
+	struct request_queue **queues;
+	int n_queues;
+	int size;
+};
+
+struct queue_arr all = {0};
```

Comment:
Shouldn't it be static? And zero initialization is meaningless for global variables, because they are zero initialized by default. And why aren't you just using rb_tree instead of resizable array of pointers?